miio/heater_miot.py

```diff
@@ -0,0 +1,228 @@
+    "temperature_range": (18, 28),
+    "delay_off_range": (0, 12 * 3600),
+}
+
+
+class LedBrightness(enum.Enum):
+    On = 0
+    Off = 1
+
+
+class HeaterMiotException(DeviceException):
+    pass
+
+
+class HeaterMiotStatus:
+    """Container for status reports from the Xiaomi Smart Space Heater S."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        """
+        Response (MIoT format) of Xiaomi Smart Space Heater S (zhimi.heater.mc2):
+
+        [
+          { "did": "power", "siid": 2, "piid": 1, "code": 0, "value": False },
+          { "did": "target_temperature", "siid": 2, "piid": 5, "code": 0, "value": 18 },
+          { "did": "countdown_time", "siid": 3, "piid": 1, "code": 0, "value": 0 },
+          { "did": "temperature", "siid": 4, "piid": 7, "code": 0, "value": 22.6 },
+          { "did": "child_lock", "siid": 5, "piid": 1, "code": 0, "value": False },
+          { "did": "buzzer", "siid": 6, "piid": 1, "code": 0, "value": False },
+          { "did": "led_brightness", "siid": 7, "piid": 3, "code": 0, "value": 0 }
+        ]
+        """
+        self.data = data
+
+    @property
+    def power(self) -> str:
+        """Power state."""
+        return "on" if self.is_on else "off"
+
+    @property
+    def is_on(self) -> bool:
+        """True if device is currently on."""
+        return self.data["power"]
+
+    @property
+    def target_temperature(self) -> int:
+        """Target temperature."""
+        return self.data["target_temperature"]
+
+    @property
+    def delay_off_countdown(self) -> int:
+        """Countdown until turning off in seconds."""
+        return self.data["countdown_time"]
+
+    @property
+    def temperature(self) -> float:
+        """Current temperature."""
+        return self.data["temperature"]
+
+    @property
+    def child_lock(self) -> bool:
+        """True if child lock is on, False otherwise."""
+        return self.data["child_lock"] is True
+
+    @property
+    def buzzer(self) -> bool:
+        """True if buzzer is turned on, False otherwise."""
+        return self.data["buzzer"] is True
+
+    @property
+    def led_brightness(self) -> LedBrightness:
+        """LED indicator brightness."""
+        return LedBrightness(self.data["led_brightness"])
+
+    def __repr__(self) -> str:
+        s = (
+            "<HeaterMiotStatus power=%s, "
+            "target_temperature=%s, "
+            "temperature=%s, "
+            "led_brightness=%s, "
+            "buzzer=%s, "
+            "child_lock=%s, "
+            "delay_off_countdown=%s "
+            % (
+                self.power,
+                self.target_temperature,
+                self.temperature,
+                self.led_brightness,
+                self.buzzer,
+                self.child_lock,
+                self.delay_off_countdown,
+            )
+        )
+        return s
+
+
+class HeaterMiot(MiotDevice):
+    """Main class representing the Xiaomi Smart Space Heater S."""
```

Comment:
```suggestion
    """Main class representing the Xiaomi Smart Space Heater S (zhimi.heater.mc2)."""
```
Please add a note in this docstring about the supported model. 

At some point in the future this project will hopefully have a model<->implementation class type of mapping for automatic generation of supported devices in docs, discovery helper and potential other uses. When the time comes, having this information in the source will make the process much easier.